### PR TITLE
010 delete list item

### PR DESCRIPTION
**Pull request by felix-horn**

### CODE REVIEW

File: src/01.UI-Elements/ListItem.js [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
       <Checkbox
         type="checkbox"
         color="default"
         checked={isChecked}
         onChange={onToggle}
+        data-testid="checkbox"
       />
-      <TitleStyled>{title}</TitleStyled>
+      <TitleStyled data-testid="title-list-item">{title}</TitleStyled>
+      <DeleteIconStyled onClick={onDelete} data-testid="delete-list-item" />
```
Comment (mmurakaru) [inline src/01.UI-Elements/ListItem.js]: Even though this is an icon you could still consider naming this eg. DeleteButton, an icon is representative of sth and a button is interactive. Similar to your UndoButton.js where the UndoButtonStyled is representing a div.

File: src/01.UI-Elements/ListItem.test.js [hunk 2]
```diff
@@ -11,8 +11,14 @@ describe('ListItem', () => {
 
   it('toggles the checkbox by property change', () => {
```
Comment (mmurakaru) [inline src/01.UI-Elements/ListItem.test.js]: typo: toggles ... 'on' sth

File: src/01.UI-Elements/ListItem.test.js [hunk 3]
```diff
@@ -47,4 +60,21 @@ describe('ListItem', () => {
 
     expect(checkbox).toHaveProperty('checked', true)
   })
+  it('calls onDelete', () => {
+    const onToggleMock = jest.fn()
```
Comment (mmurakaru) [inline src/01.UI-Elements/ListItem.test.js:64]: Where do you expect the onToggleMock function to be called?

File: src/01.UI-Elements/UndoButton.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+const UndoButtonStyled = styled.div`
+  box-shadow: 0 1px 3px #0004;
+  height: 40px;
+  width: 85px;
+  border-radius: 20px;
+  background: white;
+  display: grid;
+  place-items: center;
```
Comment (mmurakaru) [inline src/01.UI-Elements/UndoButton.js:29]: place-items center is the best 👏🏼

File: src/hooks/useList.js [hunk 5]
```diff
@@ -10,34 +10,51 @@ export default function useList() {
   const [uncheckedIds, setUncheckedIds] = useState([])
   const [checkedIds, setCheckedIds] = useState([])
 
+  const [deletedListItem, setDeletedListItem] = useState({
+    title: '',
+    isChecked: false,
+  })
+  const [visibilityUndoButton, setVisibilityUndoButton] = useState('hidden')
```
Comment (mmurakaru) [inline src/hooks/useList.js:17]: Consider naming this just 'visibility'?

File: src/hooks/useList.js [hunk 6]
```diff
@@ -52,11 +69,39 @@ export default function useList() {
...
+    const isChecked = list.byId[targetId].isChecked
+    setDeletedListItem({ title, isChecked })
+
+    setVisibilityUndoButton('shown')
+    resetTimer()
+  }
+
+  function resetTimer() {
+    timer.current && clearTimeout(timer.current)
+    timer.current = setTimeout(
```
Comment (mmurakaru) [inline src/hooks/useList.js:102]: nice!

**Review by mmurakaru — COMMENTED** [review]
Looks good to me, just found some small things that you could consider ;)

File: src/01.UI-Elements/InputForm.js [hunk 7]
```diff
@@ -35,14 +36,14 @@ const InputFormStyled = styled.label`
   align-items: center;
 `
 const AddIconStyled = styled(AddIcon)`
-  color: #878889;
+  color: var(--mid-gray);
```
Comment (monroho) [inline src/01.UI-Elements/InputForm.js:39]: useful addition ;-)

File: src/01.UI-Elements/UndoButton.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+```js padded
+<UndoButton visibility="shown"/>
```
Comment (monroho) [inline src/01.UI-Elements/UndoButton.md]: visibility? Where is that declared? Couldn´t find that.

File: src/01.UI-Elements/UndoButton.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
...
+```
```
Comment (FabianAlleblas) [inline src/01.UI-Elements/UndoButton.md]: I cant find the visibility anywhere else? What is it for?

File: src/01.UI-Elements/UndoButton.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import styled from 'styled-components/macro'
+import UndoIcon from '@material-ui/icons/Undo'
+
+export default function UndoButton({ onClick, className }) {
+  return (
```
Comment (FabianAlleblas) [inline src/01.UI-Elements/UndoButton.js]: No PropTypes?

File: src/hooks/useList.js [hunk 11]
```diff
@@ -52,11 +69,39 @@ export default function useList() {
...
+      setUncheckedIds(uncheckedIds.filter((id) => id !== targetId))
       setCheckedIds([targetId, ...checkedIds])
     }
   }
+
+  function deleteListItem(targetId) {
+    const copyOfById = { ...list.byId }
+    delete copyOfById[targetId]
+    const byIdWithoutTargetId = copyOfById
+
```
Comment (monroho) [inline src/hooks/useList.js:84]: The naming is pretty hard to understand here ( maybe it`s just me )

File: src/hooks/useList.js (hunk 5 above)
Comment (monroho) [reply src/hooks/useList.js:17]: +1  
edit:
(Regarding Markus suggestion to rename it to something like "visibility" and "setVisibility" )

File: src/styles/GlobalStyle.js [hunk 12]
```diff
@@ -2,19 +2,32 @@ import { createGlobalStyle } from 'styled-components/macro'
 
 export default createGlobalStyle`
 
+:root {
+    
+
+    --almost-black: #212121; //font-color
```
Comment (FabianAlleblas) [inline src/styles/GlobalStyle.js:8]: --nearly-black

**Review by FabianAlleblas — COMMENTED** [review]
Nice, but I struggled a bit with your naming, maybe you could consider to simplify a lot of the names.

File: src/App.js [hunk 13]
```diff
@@ -2,29 +2,69 @@ import useList from './hooks/useList'
...
+    0% {
+      opacity: 1;
+      pointer-events: all;
+    }
+    100% {
+      opacity: 0;
+      pointer-events: none;
+    }
+  }
+`
```
Comment (monroho) [inline src/App.js:70]: Really nice touch with the fadeout !

Comment (felix-horn) [reply src/App.js:70]: No big deal, only took me two days and Jerry's help. Easy... :)))

File: src/hooks/useList.js (hunk 11 above)
Comment (felix-horn) [reply src/hooks/useList.js:84]: I don't like it either, but I couldn't come up with an easier solution. I even added the line `const byIdWithoutTargetId = copyOfById` in order to make it easier to understand. I also could have left that out and kept on working with `copyOfById` which doesn't include the deleted part anymore...
Do you have a better idea?

File: src/01.UI-Elements/UndoButton.md (hunk 8 above)
Comment (felix-horn) [reply src/01.UI-Elements/UndoButton.md]: Good point. Thanks! I forgot to change it here.

File: src/01.UI-Elements/UndoButton.js (hunk 10 above)
Comment (felix-horn) [reply src/01.UI-Elements/UndoButton.js]: Upsi....

File: src/01.UI-Elements/UndoButton.md (hunk 9 above)
Comment (felix-horn) [reply src/01.UI-Elements/UndoButton.md]: Good point. Thanks! I forgot to change it here.

**Pull request merged**